### PR TITLE
Decouple yaml file and KubernetesClientConfiguration object

### PR DESCRIPTION
**Pull request by tg123**
KubernetesClientConfiguration now can be created without yaml config 
first step to address #24 

Introduce go client style config creating functions

* Anonymous

```
var client = new Kubernetes(new KubernetesClientConfiguration
{
       Host = "http://127.0.0.1:8001"
});
```

* InClusterConfig
```
var k8sClientConfig = KubernetesClientConfiguration.InClusterConfig();
var client = new Kubernetes(k8sClientConfig);
```

* BuildConfigFromFlags like
```
//  BuildConfigFromConfigFile(string masterUrl = null, string kubeconfigPath = null)
var k8sClientConfig = KubernetesClientConfiguration.BuildConfigFromConfigFile();
var client = new Kubernetes(k8sClientConfig);
```

### CODE REVIEW

**Comment by krabhishek8260** [comment]
LGTM. Thanks for the PR.

File: src/Kubernetes.Auth.cs [hunk 1]
```diff
@@ -24,21 +24,24 @@ public Kubernetes(KubernetesClientConfiguration config)
...
-            var handler = new HttpClientHandler
+            if (BaseUri.Scheme == "https")
             {
-                ServerCertificateCustomValidationCallback = sslCertValidationFunc
-            };
+                if (config.SkipTlsVerify)
+                {
+                    handler.ServerCertificateCustomValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+                }
+                else
```
Comment (brendandburns) [inline src/Kubernetes.Auth.cs]: combine the `else` and `if` into an `else if`

Comment (brendandburns) [reply src/Kubernetes.Auth.cs]: oh, wait, nevermind, I see there's a statement under the if...

File: src/KubernetesClientConfiguration.InCluster.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            var host = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_HOST");
+            var port = Environment.GetEnvironmentVariable("KUBERNETES_SERVICE_PORT");
+
+            if (string.IsNullOrWhiteSpace(host) || string.IsNullOrWhiteSpace(port))
+            {
+                throw new KubeConfigException(
+                    "unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined");
+            }
+
+            var token = File.ReadAllText("/var/run/secrets/kubernetes.io/serviceaccount/" + ServiceAccountTokenKey);
```
Comment (brendandburns) [inline src/KubernetesClientConfiguration.InCluster.cs]: extract `"/var/run/secrets/kubernetes.io/serviceaccount/"` and other strings out as constants?

**Comment by brendandburns** [comment]
Needs a rebase, but then LGTM.

File: src/KubernetesClientConfiguration.InCluster.cs (hunk 2 above)
Comment (tg123) [reply src/KubernetesClientConfiguration.InCluster.cs]: another small question
Is this still not supported by windows after 1709?

**Pull request merged**